diff --git a/lib/init.js b/lib/init.js
index 82e87fac5e01285763711f0203a040e8be18bf83..8a0c53185570d06f378ec8e618ca01e30e1bf50d 100644
--- a/lib/init.js
+++ b/lib/init.js
@@ -53,6 +53,7 @@ export async function init({ authOptions, providerId, action, url, cookies: reqC
             logo: "",
             brandColor: "",
             buttonText: "",
+            customCss: "",
         },
         // Custom options override defaults
         ...authOptions,
diff --git a/lib/pages/index.js b/lib/pages/index.js
index 263b25a7159bf9456bcf8f892c6f196f84ee6c77..64e1d3e021618e7422741e00ec3826c47c33ce85 100644
--- a/lib/pages/index.js
+++ b/lib/pages/index.js
@@ -9,7 +9,7 @@ function send({ html, title, status, cookies, theme }) {
         cookies,
         status,
         headers: { "Content-Type": "text/html" },
-        body: `<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><style>${css}</style><title>${title}</title></head><body class="__next-auth-theme-${theme?.colorScheme ?? "auto"}"><div class="page">${renderToString(html)}</div></body></html>`,
+        body: `<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><style>${css}</style><style>${theme?.customCss}</style><title>${title}</title></head><body class="__next-auth-theme-${theme?.colorScheme ?? "auto"}"><div class="page">${renderToString(html)}</div></body></html>`,
     };
 }
 /**
diff --git a/src/types.ts b/src/types.ts
index 426cf47b29fe37fd082f767a23ca989690dcab63..2db2dfb6cb4b951b601eca21dfe38496aa3e389b 100644
--- a/src/types.ts
+++ b/src/types.ts
@@ -89,6 +89,7 @@ export interface Theme {
   logo?: string
   brandColor?: string
   buttonText?: string
+  customCss?: string
 }
 
 /**
diff --git a/types.d.ts b/types.d.ts
index 4e37219bb13e05b141a9be92e55b278598e1f268..3531373839d5aa3ca11813e1d3cf821389aa7146 100644
--- a/types.d.ts
+++ b/types.d.ts
@@ -75,6 +75,7 @@ export interface Theme {
     logo?: string;
     brandColor?: string;
     buttonText?: string;
+    customCss?: string;
 }
 /**
  * Different tokens returned by OAuth Providers.
