diff --git a/lib/sqlite.js b/lib/sqlite.js
index 0cbeaf3cc8a70087608d579b13555558abaaec1e..471c7ea74968b512b1e0d7776f370f4efa45c28a 100644
--- a/lib/sqlite.js
+++ b/lib/sqlite.js
@@ -109,14 +109,21 @@ export function SQLiteDrizzleAdapter(client, tableFn = defaultSqliteTableFn) {
             };
             return account;
         },
-        getUserByAccount(account) {
-            const results = client
+        // This should hopefully be patched by:
+        // https://github.com/nextauthjs/next-auth/pull/8379/
+        async getUserByAccount(account) {
+            const results = await client
                 .select()
                 .from(accounts)
                 .leftJoin(users, eq(users.id, accounts.userId))
                 .where(and(eq(accounts.provider, account.provider), eq(accounts.providerAccountId, account.providerAccountId)))
                 .get();
-            return results?.user ?? null;
+
+            if (!results) {
+                return null;
+            }
+
+            return results.user;
         },
         deleteSession(sessionToken) {
             return (client
